Assets/MixedRealityToolkit.SDK/Experimental/Features/Dwell/DwellHandler.cs

```diff
@@ -0,0 +1,243 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using System;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Dwell
+{
+    [Serializable]
+    public class PointerUnityEvent : UnityEvent<IMixedRealityPointer>
+    {
+    }
+
+    /// <summary>
+    /// Use this component to add a Dwell modality (https://docs.microsoft.com/en-us/windows/mixed-reality/gaze-and-dwell) to the UI target.
+    /// </summary>
+    public class DwellHandler : MonoBehaviour, IMixedRealityFocusHandler
+    {
+        protected enum DwellStateType
+        {
+            None = 0,
+            FocusGained,
+            DwellIntended,
```

Comment:
>DwellIntended, [](start = 12, length = 14)

It might be worth adding docomments to this enum to better explain the transitions and how they work:

i.e. the state machine is:

Current State       |     Valid Transitions
None                    |  FocusGained
FocusGained       |  None
FocusGained       |  DwellIntended
DwellIntended    |  DwellCanceled
DwellIntended    |  DwellStarted
DwellCanceled   |  None

Also in particular, it would be helpful to explain the term "DwellIntended" - the others feel like they're relatively clear

Reply:
That is a good idea. by documents do you mean in md files or in code comments ?

Reply:
Just code comments - I think that they're generally helpful here and then when we get around to doing MD files we can just XREF to these code comments

Reply:
Yeah sorry, doccomments == the <summary> type code comments that generate stuff through docfx.

comments = regular code comments
documentation markup stuff = MD file stuff

Reply:
Should i rename "DwellIntended" to "DelayedHover" ? (i have received similar feedback from other folks in MRW, that DwellIntended is not clear enough terminology)

---
In reply to: [314125680](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5594#discussion_r314125680) [](ancestors = 314125680)